project1/client/client.cpp

```diff
@@ -124,7 +124,7 @@ bool Client::FtpShell() {
     std::getline(std::cin, input);
 
     // determine command
-    ip_ = new InputParser(input);
+    std::unique_ptr<input_parser::InputParser> ip_ = std::make_unique<input_parser::InputParser>(input);
```

Comment:
This looks fine to me. Only thing is, I'd remove the trailing underscore from the name since it's a local variable and not a class member.